docs/designs/0000-template.md

```diff
@@ -0,0 +1,23 @@
+# Title
+
+Updated by: [0000-this-doc-does-not-exist]()
+
+Obsoleted by: [0000-this-doc-does-not-exist]()
+
+*(these sections should only be added when necessary, and should always include the links to the relevant documents)*
+
+## Introduction
+
+This section should bring a short summary of what this document is covering, the main points it is going to describe, and why it is important. Ideally should be kept under two paragraphs.
+
+## Context [optional]
+
+Explains the context outside the mere technical aspects of the design that existed when it was written. It must make clear what other influences where at play that affected the decisions that were taken. In case there is nothing relevant to mention about the context, this section can be skipped.
+
+## [Free-form sections]
+
+The actual content of the design should be placed under one or more sections that can be named freely. Note that they should be level 2 headings (H2), but each section can also be subdivided by nested headings (H3, H4, H5 or H6) if needed. Only a single level 1 heading (H1) should exist, and it is strictly reserved for the title.
```

Comment:
I'm not convinced there should be `1+` H2 headings for this, the design should be covered by a single H2 section and N subsections, otherwise we'll break the structure which we're trying to standardize somewhat I think.

Reply:
I agree this looks good on the template, as it is more standardized, but having a large document with many subsections forced into a H2 brings a horrible result, at least IMO. As an example, check the Ruby design in this [commit](https://github.com/containerbuildsystem/cachi2/pull/601/commits/46e89a81d206ee3221e7a6a1bf0926408a9dae4e?short_path=d9b9f40#diff-d9b9f40563b29b3b51c53bc4ec6ea3b9a63fc3fadd81c8f00bb91a535561ad39), and how I changed it in the following ones. For smaller docs, I believe a single H2 works perfectly.

I am not sure what's the best option here.  If nobody minds the excessive nesting (and the hard limit on the H5), then let's just make it a single H2. 

(I even considered a separate template for new package manager designs, although I think I still prefer to keep things simple).

Reply:
> with many subsections forced into a H2 brings a horrible result

We can agree that it's a matter of taste. On the other hand...

>  If nobody minds the excessive nesting (and the hard limit on the H5)

... if we ever need to go past H5 then I guess we need to rethink whether a given design really needs it, I don't expect that to be a regular thing, but I might be wrong here.